### PR TITLE
fix: SOF-769 adlib toggling with toggleOnSingleClick enabled

### PR DESCRIPTION
**Pull request by ianshade**
Make a single tap on a dashboard button toggle the adlib once. (Before, when `toggleOnSingleClick` was `true`, it toggled it twice, leading to a piece starting and almost immediately stopping afterwards)
Make the MiniShelf and TimelineDashboardPanel respect their filter's `toggleOnSingleClick` option.
Removes the `no-pointer` hack in favour of enabling `toggleOnSingleClick` on our dashboard panels, which was the intended way of making a single click/tap toggle adlibs (will come in a separate PR for the blueprints).

### CODE REVIEW

**Review by KvelaGorrrrnio — APPROVED** [review]
It looks good 👏

File: meteor/client/ui/Shelf/DashboardPieceButton.tsx [hunk 1]
```diff
@@ -417,8 +417,11 @@ export class DashboardPieceButtonBase<T = {}> extends MeteorReactComponent<
 	}
 
 	private handleOnPointerUp = (e: React.PointerEvent<HTMLDivElement>) => {
+		const { toggleOnSingleClick } = this.props
 		if (e.pointerId === this.pointerId) {
-			this.props.onToggleAdLib(this.props.piece, e.shiftKey || !!this.props.queueAllAdlibs, e)
+			if (!toggleOnSingleClick) {
```
Comment (LindvedKrvang) [inline meteor/client/ui/Shelf/DashboardPieceButton.tsx:422]: You can prevent the nesting by inverting the first if-statement:

```
private handleOnPointerSetup = (...) => {
  if (e.pointerId !== this.pointerId) {
    return
  }
  const { toggleOnSigleClick } = this.props
  if (!toggleOnSingleClick) {
    // your onToggleAdlib funtion here
  }
  e.preventDefault()
}
```

**Review by LindvedKrvang — APPROVED** [review]

File: meteor/client/ui/Shelf/DashboardPieceButton.tsx [hunk 2]
```diff
@@ -417,8 +417,11 @@ export class DashboardPieceButtonBase<T = {}> extends MeteorReactComponent<
 	}
 
 	private handleOnPointerUp = (e: React.PointerEvent<HTMLDivElement>) => {
+		const { toggleOnSingleClick } = this.props
```
[the remark this answers is not in the document]
Comment (ianshade) [reply meteor/client/ui/Shelf/DashboardPieceButton.tsx:420]: Are we talking performance? Object destructuring seems to not have any noticeable impact: https://www.measurethat.net/Benchmarks/ShowResult/316043. If it was bad, would rules [like this](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/destructuring-assignment.md) be recommended?

File: meteor/client/ui/RundownView.tsx [hunk 3]
```diff
@@ -1438,21 +1440,22 @@ export const RundownView = translateWithTracker<IProps, IState, ITrackedProps>((
 			const filteredUiSegmentMap: Map<SegmentId, AdlibSegmentUi> = new Map()
 			const filteredUiSegments: AdlibSegmentUi[] = []
 			let resultSourceLayerLookup: SourceLayerLookup = {}
+			let miniShelfFilter: RundownLayoutFilterBase | undefined
 			if (props.playlist && props.showStyleBase && props.studio) {
-				let filter =
+				const possibleMiniShelfFilter =
```
[the remark this answers is not in the document]
Comment (ianshade) [reply meteor/client/ui/RundownView.tsx:1445]: Sometimes it's about using the right tool. If you reviewed the PRs using VSCode, you would just be able to hover over the variable and see the inferred type.

**Pull request merged**